src/pages/index.js

```diff
@@ -68,38 +52,199 @@ const profileEditModal = new PopupWithForm(
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      ".cards__list"
+    );
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 function handleProfileEditSubmit(inputValues) {
-  console.log(inputValues);
-  userInfo.setUserInfo(inputValues);
-  editFormValidator.resetValidation(inputValues);
-  profileEditModal.close();
+  profileEditModal.renderLoading(true);
+  api
+    .updateProfileInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      profileEditModal.close();
+      editFormElement.resetValidation();
+      editFormValidator.disableSubmit();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditModal.renderLoading(false);
+    });
 }
 
 function handleAddCardFormSubmit(inputValues) {
-  renderCard(inputValues);
-  addFormValidator.resetValidation(inputValues);
+  addCardModal.renderLoading(true);
+  api
+    .createACard(inputValues)
+    .then((res) => {
+      renderCard(res);
+      addCardModal.close();
+      addFormElement.resetValidation();
+      addFormValidator.disableSubmit();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardModal.renderLoading(false);
+    });
+}
 
-  addCardModal.close();
+function handleChangeAvatar(inputValues) {
+  avatarModal.renderLoading(true);
+  api
+    .updateAvatar(inputValues)
+    .then((res) => {
+      userInfo.setAvatarInfo(res);
+      avatarModal.close();
+      avatarFormValidator.disableSubmit();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarModal.renderLoading(false);
+    });
+  console.log(inputValues);
 }
 
-const editFormValidator = new FormValidator(
-  validationSettings,
-  editFormElement
-);
-const addFormValidator = new FormValidator(validationSettings, addFormElement);
+function handleImageClick(cardData) {
+  previewImageModal.open(cardData);
+}
 
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
+function handleLikeClicks(card) {
+  if (card.isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then((newCardData) => {
+        card.isLiked = newCardData.isLiked;
+        card.renderLikes();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+  if (!card.isLiked) {
+    api
+      .likeCard(card._id)
+      .then((newCardData) => {
+        card.isLiked = newCardData.isLiked;
+        card.renderLikes();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+}
+
+function handleDeleteSubmit(card) {
+  deleteImageModal.open();
+  console.log(card._id);
+  deleteImageModal.handleDelete(() => {
+    api
```

Comment:
Needs correction: don't forget to notify the user about loading by changing the button text

Reply:
For this correction, I notice that you're mentioning changing the button text to inform the user that it is loading. However, in the initial instructions the loading text was only for the add card, edit profile, and avatar change buttons, especially since the delete image modal should remove the image almost instantly. I just wanted to double check that this was correct, though. Thank you!

Reply:
The loading text should be shown for every api request (form submission), so the user will know that something is loading now

Reply:
Hey, thank you for the clarification on this as well. I made sure to fix this issue. Thank you!